server/plugins/api/GenericNotifier.ts

```diff
@@ -0,0 +1,7 @@
+import { GenericConfig } from "./GenericConfig";
+import { APIReturn } from "./APIReturn";
+
+export interface GenericNotifier {
+	sendMessage<T extends GenericConfig>(config: T): APIReturn;
+	TAG: string;
+}
```

Comment:
This should also include a method that lets plugins do a one-time setup.
So they can put in an API key or do some other checks.
It would be cool if it also had a function to check the config and see if it was valid before sending (e.g. does a slack channel exist) to prevent typos.

```typescript
export interface GenericNotifier<T> {
    async setup(): Promise<void>;
    async check(config: T): Promise<boolean>;
}
```

Reply:
typescript doesn't like the async declartaion on interfaces.

Reply:
Also, couldn't sendMessage do everything check could do? And return error status if malformed config?